index.js

```diff
@@ -153,6 +157,23 @@ class FastifyOtelInstrumentation extends InstrumentationBase {
           return
         }
 
+        if (routeOptions.otel === false) {
+          instrumentation.logger.debug(
+            `Ignoring route instrumentation ${routeOptions.method} ${routeOptions.url} because it is disabled`
+          )
+
+          // Needed for the onRequest hook to skip the request (must run first)
+          if (routeOptions.onRequest != null) {
+            routeOptions.onRequest = Array.isArray(routeOptions.onRequest)
+              ? [skipRequestHook, ...routeOptions.onRequest]
+              : [skipRequestHook, routeOptions.onRequest]
+          } else {
+            routeOptions.onRequest = skipRequestHook
```

Comment:
Why do we need to run a noop function?
We could just `return` instead of adding this hook and in the `instance.addHook('onRequest', function (request, _reply, hookDone) {` hook (WHY don't we write a function name?! 😭) we could replace `request[kSkipRequest] === true` with `request.routeOptions.config`

`